### ISSUE TITLE
ocilib::Lob::GetConnection() does not compile anymore

### ISSUE THREAD

**Issue by unficyp** [vrogier/ocilib#325]
Hi,
just upgraded to 4.7.5, recompiling my test projects says:
`In file included from /usr/local/include/ocilib.hpp:196:
/usr/local/include/ocilibcpp/detail/Lob.hpp:196:22: error: 'GetEnvironmentHandle' is a private member of 'ocilib::Environment'
        Environment::GetEnvironmentHandle()
                     ^
/usr/local/include/ocilibcpp/types.hpp:1141:30: note: declared private here
        static core::Handle* GetEnvironmentHandle();
                             ^
1 error generated.`

do i miss something?

regards

**Comment by vrogier** [comment]
It seems indeed that a friend class declaration for Lob<T,U> is missing in Environment class declaration ....
I do not understand why it ran fine on my environment :(

I will check this later today.

Can you add the following in include/ocilibcpp/ types.hpp, near line 511:

```c++
template<class, int>
friend class Lob;
```

it should get rif off the issue.

regards,

Vincent

**Comment by unficyp** [comment]
that gives me
`In file included from /usr/local/include/ocilib.hpp:153:
/usr/local/include/ocilibcpp/types.hpp:499:22: error: template 'Lob' cannot be referenced with a class specifier
        friend class Lob;
                     ^
/usr/local/include/ocilibcpp/config.hpp:198:7: note: declared here
class Lob;
      ^`

**Comment by vrogier** [comment]
Sorry, I gave you the hint while I was away from the keyboard.

Correct syntax is:

```
 template<class, int>
friend class Lob;
```

**Comment by vrogier** [comment]
and the reason why I did not notice the error is that Lob::GetConnection() is not called din my test suite :(

Will be be fixed in v4.7.6 branch.

Regards,

Vincent

**Comment by vrogier** [comment]
Hi, 

v4.7.6 released.

Regards,

Vincent
